actors/builtin/miner/miner_actor.go

```diff
@@ -2222,29 +2231,38 @@ func requestUnsealedSectorCID(rt Runtime, proofType abi.RegisteredSealProof, dea
 	return cid.Cid(unsealedCID)
 }
 
-func requestDealWeight(rt Runtime, dealIDs []abi.DealID, sectorStart, sectorExpiry abi.ChainEpoch) market.VerifyDealsForActivationReturn {
-	if len(dealIDs) == 0 {
-		return market.VerifyDealsForActivationReturn{
-			DealWeight:         big.Zero(),
-			VerifiedDealWeight: big.Zero(),
+func requestDealWeights(rt Runtime, sectors []market.SectorDeals) *market.VerifyDealsForActivationReturn {
+	// Short-circuit if there are no deals in any of the sectors.
+	dealCount := 0
+	for _, sector := range sectors {
+		dealCount += len(sector.DealIDs)
+	}
+	if dealCount == 0 {
```

Comment:
Ensuring that we don't contact market actor when we have no deals at all, but not worrying about individual sectors that contain no deals only makes sense if sending a message costs a lot of gas but message size of an internal message costs much much less. I believe this is the case. 

If message size is an issue, it would be better to move this conditional outside this method, or to move the allocation of deals into market.SectorDeals inside.

Reply:
I considered dropping the optimization altogether, but didn't want to incur the significant cost of message dispatch in the very common case of no deals. I'm less concerned about optimizing beyond that unless profiling shows it to matter.